### PR TITLE
test: use specific docker image tag in all tests

### PR DESCRIPTION
**Pull request by aanm**
Also modified ImagePullPolicy to IfNotPresent to avoid pulling the
images when creating a new container.

Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5282)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: test/runtime/Policies.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ var _ = Describe("RuntimePolicyEnforcement", func() {
 			ExpectEndpointSummary(vm, helpers.Disabled, 0)
 
 			By("Create a new container")
-			vm.ContainerCreate("new", "cilium/demo-httpd", helpers.CiliumDockerNetwork, "-l id.new")
+			vm.ContainerCreate("new", "docker.io/cilium/demo-httpd:latest", helpers.CiliumDockerNetwork, "-l id.new")
```
Comment (eloycoto) [inline test/runtime/Policies.go]: now that you change this one, could you use `helpers.HttpdImage`

File: test/runtime/Policies.go [hunk 2]
```diff
@@ -202,7 +202,7 @@ var _ = Describe("RuntimePolicyEnforcement", func() {
 			ExpectEndpointSummary(vm, helpers.Enabled, 0)
 			ExpectEndpointSummary(vm, helpers.Disabled, 1)
 
-			vm.ContainerCreate("new", "cilium/demo-httpd", helpers.CiliumDockerNetwork, "-l id.new")
+			vm.ContainerCreate("new", "docker.io/cilium/demo-httpd:latest", helpers.CiliumDockerNetwork, "-l id.new")
```
Comment (eloycoto) [inline test/runtime/Policies.go]: now that you change this one, could you use `helpers.HttpdImage`

**Review by eloycoto — CHANGES_REQUESTED** [review]

**Review by ianvernon — APPROVED** [review]
Good change overall. As part of our `precheck`, I think we should add a script which ensures that the `imagePullPolicy` is always `IfNotPresent` in `test/k8sT/manifests`. We can add this to ensure that we never add additional YAMLs which do not contain this `imagePullPolicy`.

**Comment by ianvernon** [comment]
Created #5287 to track the above.

File: test/runtime/Policies.go (hunk 2 above)
Comment (aanm) [reply test/runtime/Policies.go]: Done

File: test/runtime/Policies.go (hunk 1 above)
Comment (aanm) [reply test/runtime/Policies.go]: Done

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-upstream-k8s

**Comment by aanm** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

**Pull request merged**